src/main/java/mat/server/service/impl/FhirMeasureRemoteCallImpl.java

```diff
@@ -0,0 +1,155 @@
+    private final RestTemplate restTemplate;
+    private final FhirContext fhirContext;
+
+    @Data
+    @NoArgsConstructor
+    @AllArgsConstructor
+    public static class MeasurePackageFullData {
+            String includeBundle;
+            String measure;
+            String library;
+    }
+
+    public FhirMeasureRemoteCallImpl(RestTemplate restTemplate,FhirContext fhirContext) {
+        this.restTemplate = restTemplate;
+        this.fhirContext = fhirContext;
+    }
+
+    @Override
+    public String push(String measureId) {
+        Map<String, Object> uriVariables = new HashMap<>();
+        uriVariables.put("id", measureId);
+        return rest(fhirServicesUrl + FHIR_PUSH_MEASURE_PARAMS,
+                HttpMethod.POST,
+                String.class,
+                uriVariables);
+    }
+
+    @Override
+    public ConversionResultDto convert(String measureId, String vsacGrantingTicket, boolean draft) {
+        return convert(measureId, ConversionType.CONVERSION, vsacGrantingTicket, draft);
+    }
+
+    @Override
+    public ConversionResultDto validate(String measureId, String vsacGrantingTicket, boolean draft) {
+        return convert(measureId, ConversionType.VALIDATION, vsacGrantingTicket, draft);
+    }
+
+    @Override
+    public FhirMeasurePackageResult packageMeasure(String measureId) {
+        FhirMeasurePackageResult result = new FhirMeasurePackageResult();
+
+        Map<String, Object> uriVariables = new HashMap<>();
+        uriVariables.put("id", measureId);
+        MeasurePackageFullData packagingResp = rest(fhirServicesUrl + FHIR_PACKAGE_MEASURE_PARAMS,
+                HttpMethod.GET,
+                MeasurePackageFullData.class,
+                uriVariables);
+
+        result.setMeasureJson(packagingResp.getMeasure());
+        result.setInludedLibsJson(packagingResp.getIncludeBundle());
+        result.setMeasureLibJson(packagingResp.getLibrary());
+
+        Measure measure = fhirContext.newJsonParser().parseResource(Measure.class,result.getMeasureJson());
+        result.setMeasureXml(fhirContext.newXmlParser().encodeResourceToString(measure));
+
+        Library lib = fhirContext.newJsonParser().parseResource(Library.class,result.getMeasureLibJson());
+        result.setMeasureLibXml(fhirContext.newXmlParser().encodeResourceToString(lib));
+        lib.getContent().forEach(a -> {
+            if (StringUtils.equalsIgnoreCase("text/cql", a.getContentType())) {
+                result.setMeasureLibCql(decodeBase64(a));
+            } else if (StringUtils.equalsIgnoreCase("application/elm+json", a.getContentType())) {
+                result.setMeasureLibElmJson(decodeBase64(a));
+            } else if (StringUtils.equalsIgnoreCase("application/elm+xml", a.getContentType())) {
+                result.setMeasureLibElmXml(decodeBase64(a));
+            }
+        });
+
+        Bundle bundle = fhirContext.newJsonParser().parseResource(Bundle.class,result.getInludedLibsJson());
```

Comment:
Minor item: Could the xml and json parsers be re-used?

Reply:
Changed it to reuse them.